src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -226,10 +226,33 @@ def create_connect_args(self, url):
             opts["host"] = opts["host"] + ".snowflakecomputing.com"
             opts["port"] = "443"
         opts["autocommit"] = False  # autocommit is disabled by default
-        opts.update(url.query)
+
+        query = dict(**url.query)  # make mutable
+        cache_column_metadata = query.pop("cache_column_metadata", None)
         self._cache_column_metadata = (
-            opts.get("cache_column_metadata", "false").lower() == "true"
+            parse_url_boolean(cache_column_metadata) if cache_column_metadata else False
         )
+
+        for name, value in query.items():
```

Comment:
I don't think I understand what this for-loop is doing, can you elaborate?

Reply:
We grab the parameter to type mapping used in validation of the driver arguments to figure out what are the expected types of each argument - if it's a boolean, we then parse it from string, otherwise pass it down as a string to the driver.

Reply:
Thanks for the clarification. Let's maybe add it into the comment? :)